### PR TITLE
Xp-161 SDK code documenation

### PR DESCRIPTION
**Pull request by janpetschexain**
### References

- [XP-161](https://xainag.atlassian.net/browse/XP-161)

### Summary

- add variable typing
- improve variable naming
- update function docstrings
- add inline comments
- restructure proto and grpc directories

### Are there any open tasks/blockers for the ticket?

- none

---

### Reviewer checklist

*Reviewer agreement:*

* Reviewers assign themselves at the start of the review.
* Reviewers do **not** commit or merge the merge request.
* Reviewers have to check and mark items in the checklist.

**Merge request checklist**

- [x] Conforms to the merge request title naming `XP-XXX <a description in imperative form>`.
- [x] Each commit conforms to the naming convention `XP-XXX <a description in imperative form>`.
- [x] Linked the ticket in the merge request title or the references section.
- [x] Added an informative merge request summary.

**Code checklist**

- [x] Conforms to the branch naming `XP-XXX-<a_small_stub>`.
- [x] Passed scope checks.
- [x] Added or updated tests if needed.
- [x] Added or updated code documentation if needed.
- [x] Conforms to Google docstring style.
- [x] Conforms to XAIN structlog style.

### CODE REVIEW

File: xain_sdk/logger.py [hunk 1]
```diff
@@ -1,24 +1,44 @@
-"""This module contains all the xain logging configuration"""
-
+"""Logging configuration."""
 
 import logging
+from typing import Type
+
+import structlog
```
Comment (Robert-Steiner) [inline xain_sdk/logger.py]: structlog is missing in the setup.py

File: xain_sdk/grpc/participant.py [hunk 2]
```diff
@@ -9,70 +9,82 @@
...
-    TRAINING = auto()
-    POST_TRAINING = auto()
-    DONE = auto()
+    WAITING_FOR_SELECTION: auto = auto()
+    TRAINING: auto = auto()
+    POST_TRAINING: auto = auto()
+    DONE: auto = auto()
 
 
 def rendezvous(channel: grpc.Channel) -> None:
```
Comment (PanicButtonPressed) [inline xain_sdk/grpc/participant.py]: `from grpc import Channel`

File: xain_sdk/grpc/participant.py [hunk 3]
```diff
@@ -9,70 +9,82 @@
...
+    """Start a rendezvous exchange with a coordinator.
 
     Args:
-        channel: gRPC channel to Coordinator.
+        channel (~grpc.Channel): A gRPC channel to the coordinator.
     """
-    stub: CoordinatorStub = CoordinatorStub(channel)  # type: ignore
 
-    response = coordinator_pb2.RendezvousResponse.LATER
+    coordinator: CoordinatorStub = CoordinatorStub(channel=channel)  # type: ignore
```
Comment (PanicButtonPressed) [inline xain_sdk/grpc/participant.py]: unused type ignore?

File: xain_sdk/logger.py [hunk 4]
```diff
@@ -1,24 +1,44 @@
...
+from structlog._config import BoundLoggerLazyProxy
 
+StructLogger: Type[BoundLoggerLazyProxy] = BoundLoggerLazyProxy
```
Comment (PanicButtonPressed) [inline xain_sdk/logger.py]: Unfortunately, this is resulting in mypy error in the `participant_state_machine`module: `Variable "xain_sdk.logger.StructLogger" is not valid as a type`

File: xain_sdk/grpc/participant.py [hunk 5]
```diff
@@ -9,70 +9,82 @@
...
 
-    while response == coordinator_pb2.RendezvousResponse.LATER:
-        reply = stub.Rendezvous(coordinator_pb2.RendezvousRequest())
-        if reply.response == coordinator_pb2.RendezvousResponse.ACCEPT:
-            print("Participant received: ACCEPT")
-        elif reply.response == coordinator_pb2.RendezvousResponse.LATER:
-            print(f"Participant received: LATER. Retrying in {RETRY_TIMEOUT}s")
-            time.sleep(RETRY_TIMEOUT)
+    response: RendezvousResponse = RendezvousResponse.Value("LATER")
```
Comment (PanicButtonPressed) [inline xain_sdk/grpc/participant.py]: With a newer version of grpc, the functionality was introduced to directly address the value of an enum attribute by accessing the value attribute like `some_enum_name.attribute`. As stated in the protocol buffer documentation this is the most familiar one, https://developers.google.com/protocol-buffers/docs/pythontutorial#enums. However, I'll let it up to you, which one we'll use.

File: xain_sdk/logger.py (hunk 1 above)
Comment (janpetschexain) [reply xain_sdk/logger.py]: added

File: setup.py [hunk 6]
```diff
@@ -99,10 +99,7 @@ def run(self):
     long_description=readme,
     long_description_content_type="text/markdown",
     url="https://github.com/xainag/xain-sdk",
-    author=[
-        "Daniel J. Beutel <daniel.beutel@xain.io>",
-        "Taner Topal <taner.topal@xain.io>",
-    ],
+    author=["Daniel J. Beutel <daniel.beutel@xain.io>", "Taner Topal <taner.topal@xain.io>",],
```
[the remark this answers is not in the document]
Comment (janpetschexain) [reply setup.py]: changed

File: xain_sdk/grpc/participant.py (hunk 2 above)
Comment (janpetschexain) [reply xain_sdk/grpc/participant.py]: changed

File: xain_sdk/logger.py (hunk 4 above)
Comment (janpetschexain) [reply xain_sdk/logger.py]: changed. it's a pity that typing is not possible in this way for type aliases.

File: xain_sdk/grpc/participant.py (hunk 5 above)
Comment (janpetschexain) [reply xain_sdk/grpc/participant.py]: changed. nice feature!

File: xain_sdk/grpc/participant.py (hunk 3 above)
Comment (janpetschexain) [reply xain_sdk/grpc/participant.py]: removed. this might be necessary again once the ci is in place (cf. `calling untyped function in a typed context`).

**Comment by janpetschexain** [comment]
> Get the following error when I start the participant
> 
> ```
> Exception in thread Thread-1:
> Traceback (most recent call last):
>   File "/Users/robertsteiner/.pyenv/versions/3.7.4/lib/python3.7/threading.py", line 926, in _bootstrap_inner
>     self.run()
>   File "/Users/robertsteiner/.pyenv/versions/3.7.4/lib/python3.7/threading.py", line 870, in run
>     self._target(*self._args, **self._kwargs)
>   File "/Users/robertsteiner/projects/xain-sdk/xain_sdk/participant_state_machine.py", line 276, in message_loop
>     time.sleep(secs=HEARTBEAT_TIME)
> TypeError: sleep() takes no keyword arguments
> ```

fixed.

**Pull request merged**